Sources/Talkshoplive/Core/Chat/ChatProvider.swift

```diff
@@ -438,6 +439,15 @@ public class ChatProvider {
                         // Notify when all message processing is complete
                         dispatchGroup.notify(queue: .main) {
                            Config.shared.isDebugMode() ? print("History : Fetched successfully!") : ()
+                            //Sort data according to published timeinterval
+                            messageArray.sort(by: {
+                                guard let publishedFirst = $0.published, let timeIntervalFirst = Int(publishedFirst),
```

Comment:
Messages should be in desc order by default when you pass the start timetoken.

Reply:
@daman-talkshoplive it's coming in proper order , confirmed with front end as well.